shared/profile/index.js

```diff
@@ -4,28 +4,72 @@ import {connect} from 'react-redux'
 import Render from './render'
 import type {Props} from './render'
 import flags from '../util/feature-flags'
+import {getProfile, updateTrackers} from '../actions/tracker'
+import {routeAppend, navigateUp} from '../actions/router'
 
 class Profile extends Component<void, Props, void> {
-  static parseRoute () {
+  static parseRoute (currentPath, uri) {
     return {
-      componentAtTop: {title: 'Profile'},
+      componentAtTop: {
+        title: 'Profile',
+        props: {
+          username: currentPath.get('username'),
+          profileRoot: uri.count() && uri.last().get('path') === 'root',
```

Comment:
Perhaps call this `profileIsRoot`? It was unclear from the name that this was a boolean flag.